src/goto-programs/resolve_inherited_component.cpp

```diff
@@ -58,10 +52,16 @@ resolve_inherited_componentt::operator()(
       return inherited_componentt(current_class, component_name);
     }
 
-    const auto current_class_id = class_hierarchy.class_map.find(current_class);
-    if(current_class_id != class_hierarchy.class_map.end())
+    const auto current_class_symbol_it =
+      symbol_table.symbols.find(current_class);
+
+    if(current_class_symbol_it != symbol_table.symbols.end())
     {
-      const class_hierarchyt::idst &parents = current_class_id->second.parents;
+      const std::vector<irep_idt> parents =
+        make_range(to_struct_type(current_class_symbol_it->second.type).bases())
+          .map([](const struct_typet::baset &base) {
+            return base.type().get_identifier();
```

Comment:
:bulb: the vector is not strictly necessary here, you could have used `auto` to have it as a range instead